src/System.Runtime.Extensions/tests/System/IO/Path.cs

```diff
@@ -127,12 +198,104 @@ public static void GetTempFileName()
+            Assert.Throws<NotSupportedException>(() => Path.GetFullPath(@"bad:path"));
+
+            // Some Windows-specific bad paths
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(Path.DirectorySeparatorChar + ".. ." + Path.DirectorySeparatorChar));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(Path.DirectorySeparatorChar + ". ." + Path.DirectorySeparatorChar));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(Path.DirectorySeparatorChar + " ." + Path.DirectorySeparatorChar));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath("C:..."));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C:...\somedir"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C  :"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C  :\somedir"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"bad::$DATA"));
+            Assert.Throws<PathTooLongException>(() => Path.GetFullPath(@"C:\" + new string('a', 255) + @"\"));
+
+            // Some Windows-specific strange but legal paths
+            Assert.Equal(
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar),
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar + ". " + Path.DirectorySeparatorChar));
+            Assert.Equal(
+                Path.GetFullPath(Path.GetDirectoryName(curDir) + Path.DirectorySeparatorChar),
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar + "..." + Path.DirectorySeparatorChar));
+            Assert.Equal(
+                Path.GetFullPath(Path.GetDirectoryName(curDir) + Path.DirectorySeparatorChar),
+                Path.GetFullPath(curDir + Path.DirectorySeparatorChar + ".. " + Path.DirectorySeparatorChar));
+
+            // Windows-specific UNC paths
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@"\\server\share"));
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@" \\server\share"));
+            Assert.Equal(@"\\server\share\dir", Path.GetFullPath(@"\\server\share\dir"));
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@"\\server\share\."));
+            Assert.Equal(@"\\server\share", Path.GetFullPath(@"\\server\share\.."));
+            Assert.Equal(@"\\server\share\", Path.GetFullPath(@"\\server\share\    "));
+            Assert.Equal(@"\\server\  share\", Path.GetFullPath(@"\\server\  share\"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\server"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\server\"));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\server\.."));
+            Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"\\?\GLOBALROOT\"));
+
+            // Windows short paths
```

Comment:
Sorry, took a few to find this. It is unclear whether drive specific settings are trumped by overall disable (registry). Here is what I know of:

_Short path policy reg key:_
https://technet.microsoft.com/en-us/library/cc778996.aspx

_Querying for short path availability on a drive:_
https://msdn.microsoft.com/en-us/library/windows/hardware/ff545496.aspx

fsutil has the ability to set/query this stuff. It is something we want to figure out as we can probably save quite a bit of time on the normalization paths by skipping expansion in GetFullPath().